### PR TITLE
Added new cmdlets for BillingAccounts, BillingProfiles, InvoiceSections and updated Invoicing cmdlet

### PR DESCRIPTION
**Pull request by asarkar84**
## Description

Added new cmdlets for BillingAccounts, BillingProfiles, InvoiceSections and updated Invoicing cmdlet
Deprecated few cmdlets which are no longer supported
Created following issue for the review: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/647

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Billing/Billing/Az.Billing.psd1
Comment (VeryEarly) [inline src/Billing/Billing/Az.Billing.psd1:6]: please do not modify this field

File: src/Billing/Billing/Az.Billing.psd1
Comment (VeryEarly) [inline src/Billing/Billing/Az.Billing.psd1:15]: please revert back to previous version, this will be upgraded during release by Azure-PowerShell team

File: src/Billing/Billing/Az.Billing.psd1
Comment (VeryEarly) [inline src/Billing/Billing/Az.Billing.psd1:56]: please revert this back to 1.9.0

File: src/Billing/Billing/Az.Billing.psd1
Comment (VeryEarly) [inline src/Billing/Billing/Az.Billing.psd1:125]: please revert this back

File: src/Billing/Billing/Cmdlets/BillingAccounts/GetAzureRmBillingAccount.cs
Comment (VeryEarly) [inline src/Billing/Billing/Cmdlets/BillingAccounts/GetAzureRmBillingAccount.cs:42]: use "ExpandBillingProfile" instead of "ExpandBillingProfiles"

File: src/Billing/Billing/Cmdlets/BillingAccounts/GetAzureRmBillingAccount.cs
Comment (VeryEarly) [inline src/Billing/Billing/Cmdlets/BillingAccounts/GetAzureRmBillingAccount.cs:45]: use "ExpandInvoiceSections" instead of "ExpandInvoiceSections"

File: src/Billing/Billing/Cmdlets/BillingProfiles/GetAzureRmBillingProfile.cs
Comment (VeryEarly) [inline src/Billing/Billing/Cmdlets/BillingProfiles/GetAzureRmBillingProfile.cs:41]: use "ExpandInvoiceSection" instead of "ExpandInvoiceSections"

File: src/Billing/Billing/Properties/AssemblyInfo.cs
Comment (VeryEarly) [inline src/Billing/Billing/Properties/AssemblyInfo.cs:49]: please revert to previous version

File: src/Billing/Billing/Az.Billing.psd1
Comment (asarkar84) [reply src/Billing/Billing/Az.Billing.psd1]: Sure!

File: src/Billing/Billing/Az.Billing.psd1
Comment (asarkar84) [reply src/Billing/Billing/Az.Billing.psd1]: Sure!

File: src/Billing/Billing/Az.Billing.psd1
Comment (asarkar84) [reply src/Billing/Billing/Az.Billing.psd1]: Sure!

File: src/Billing/Billing/Az.Billing.psd1
Comment (asarkar84) [reply src/Billing/Billing/Az.Billing.psd1]: Sure!

File: src/Billing/Billing/Properties/AssemblyInfo.cs
Comment (asarkar84) [reply src/Billing/Billing/Properties/AssemblyInfo.cs]: Sure!

File: src/Billing/Billing/Cmdlets/BillingProfiles/GetAzureRmBillingProfile.cs
Comment (asarkar84) [reply src/Billing/Billing/Cmdlets/BillingProfiles/GetAzureRmBillingProfile.cs]: Sure! Done.

File: src/Billing/Billing/Cmdlets/BillingAccounts/GetAzureRmBillingAccount.cs
Comment (asarkar84) [reply src/Billing/Billing/Cmdlets/BillingAccounts/GetAzureRmBillingAccount.cs]: Sure! Done.

File: src/Billing/Billing/Cmdlets/BillingAccounts/GetAzureRmBillingAccount.cs
Comment (asarkar84) [reply src/Billing/Billing/Cmdlets/BillingAccounts/GetAzureRmBillingAccount.cs]: Sure! Done. Using "ExpandInvoiceSection".

**Comment by asarkar84** [comment]
> breaking changes are not allowed in non-breaking change release, please add these deprecated logic back, keep deprecation attribute: https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-attribute-help.md#when-there-is-no-replacement-cmdlet

In the current version of SDK, we are not supporting the Get-AzBillingPeriod and Get-AzEnrollmentAccount cmdlets. USers were consuming these to get the UsageDetails which is part of the Consumption cmdlets. Not Consumption cmdlets has alternate ways to get the usage without the Billing Period etc.
Please make this as part of the breaking change release.

**Comment by VeryEarly** [comment]
> > breaking changes are not allowed in non-breaking change release, please add these deprecated logic back, keep deprecation attribute: https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-attribute-help.md#when-there-is-no-replacement-cmdlet
> 
> In the current version of SDK, we are not supporting the Get-AzBillingPeriod and Get-AzEnrollmentAccount cmdlets. USers were consuming these to get the UsageDetails which is part of the Consumption cmdlets. Not Consumption cmdlets has alternate ways to get the usage without the Billing Period etc.
> Please make this as part of the breaking change release.

Hi @asarkar84 ,

Next breaking change release should be this September, do you mean you want this PR to go with it?

**Comment by asarkar84** [comment]
@VeryEarly I have added back the Get-AzBillingPeriod and Get-AzEnrollmentAccount cmdlets. Removed CmdletDeprecation attribute from Get-AzEnrollmentAccount since we will keep supporting it. Keeping CmdletDeprecation attribute in Get-AzBillingPeriod cmdlet since this should be depricated in the next versions.
Please take a look.

**Comment by asarkar84** [comment]
> > > breaking changes are not allowed in non-breaking change release, please add these deprecated logic back, keep deprecation attribute: https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-attribute-help.md#when-there-is-no-replacement-cmdlet
> > 
> > 
> > In the current version of SDK, we are not supporting the Get-AzBillingPeriod and Get-AzEnrollmentAccount cmdlets. Users were consuming these to get the UsageDetails which is part of the Consumption cmdlets. Now Consumption cmdlets has alternate ways to get the usage without the Billing Period etc.
> > Please make this as part of the breaking change release.
> 
> Hi @asarkar84 ,
> 
> Next breaking change release should be this September, do you mean you want this PR to go with it?

We are already in the end of August end. We are OK for releasing it with breaking change version in September. What is the exact breaking change release date?

File: src/Billing/Billing/ChangeLog.md
Comment (VeryEarly) [inline src/Billing/Billing/ChangeLog.md:23]: please put release notes under
## Upcoming Release

and add "* Updated assembly version of consumption cmdlets" back under ## Version 1.0.3

**Comment by VeryEarly** [comment]
> > > > breaking changes are not allowed in non-breaking change release, please add these deprecated logic back, keep deprecation attribute: https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-attribute-help.md#when-there-is-no-replacement-cmdlet
> > > 
> > > 
> > > In the current version of SDK, we are not supporting the Get-AzBillingPeriod and Get-AzEnrollmentAccount cmdlets. Users were consuming these to get the UsageDetails which is part of the Consumption cmdlets. Now Consumption cmdlets has alternate ways to get the usage without the Billing Period etc.
> > > Please make this as part of the breaking change release.
> > 
> > 
> > Hi @asarkar84 ,
> > Next breaking change release should be this September, do you mean you want this PR to go with it?
> 
> Please note that we have added the deprecated logic back, do we still need this to be part of the breaking change release? What is the exact breaking change release date?

Hi @asarkar84 ,

This PR contains new cmdlets only now, so no need to wait for the breaking change release.

Please resolve comment and static analysis issue and ti should be good to go.

File: src/Billing/Billing/ChangeLog.md
Comment (asarkar84) [reply src/Billing/Billing/ChangeLog.md:21]: Sure! I have updated the ChangeLog and sent a new iteration.

**Comment by asarkar84** [comment]
> > > > > breaking changes are not allowed in non-breaking change release, please add these deprecated logic back, keep deprecation attribute: https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-attribute-help.md#when-there-is-no-replacement-cmdlet
> > > > 
> > > > 
> > > > In the current version of SDK, we are not supporting the Get-AzBillingPeriod and Get-AzEnrollmentAccount cmdlets. Users were consuming these to get the UsageDetails which is part of the Consumption cmdlets. Now Consumption cmdlets has alternate ways to get the usage without the Billing Period etc.
> > > > Please make this as part of the breaking change release.
> > > 
> > > 
> > > Hi @asarkar84 ,
> > > Next breaking change release should be this September, do you mean you want this PR to go with it?
> > 
> > 
> > Please note that we have added the deprecated logic back, do we still need this to be part of the breaking change release? What is the exact breaking change release date?
> 
> Hi @asarkar84 ,
> 
> This PR contains new cmdlets only now, so no need to wait for the breaking change release.
> 
> Please resolve comment and static analysis issue and ti should be good to go.

@VeryEarly I have fixed the security tests.

**Comment by asarkar84** [comment]
Hi @VeryEarly We have removed the properties DownloadUrlExpiry, Type and BillingPeriodNames from the response of Get-AzBillingInvoice cmdlet which may be considered as a breaking change. Please take a look.
We may wait till the next Breaking change release in September.

**Comment by asarkar84** [comment]
@VeryEarly Since we are very close to the October breaking change release, I would like to ensure that the below PR is good to go. Please take a look again and let me know if you expect me to take any action.

**Comment by asarkar84** [comment]
@VeryEarly Please merge this PR if it looks good now.

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**
